### PR TITLE
Remove webkit1 support

### PR DESCRIPTION
**Pull request by keis**
ifdef sections removed by running
unifdef -m -DUSE_WEBKIT2 `find src -name '*.[ch]'`

Makefile updated

Ref #182

### CODE REVIEW

**Comment by keis** [comment]
@mathstuf do you want to check this before merge?

**Comment by mathstuf** [comment]
Ah, yeah, I'll look at it this week.

File: Makefile [hunk 1]
```diff
@@ -14,45 +14,28 @@ LIBDIR     ?= $(INSTALLDIR)/lib$(LIB_SUFFIX)/uzbl
 RUN_PREFIX ?= $(PREFIX)
 INSTALL    ?= install -p
 
-ENABLE_WEBKIT2 ?= no
+ENABLE_WEBKIT2 ?= yes
 ENABLE_GTK3    ?= auto
```
Comment (mathstuf) [inline Makefile]: WebKit2 requires GTK3, so this can also be removed.

File: src/commands.c [hunk 2]
```diff
@@ -45,12 +42,10 @@ struct _UzblCommands {
     gboolean         search_forward;
     gchar           *search_text;
 
-#ifdef USE_WEBKIT2
 #if WEBKIT_CHECK_VERSION (2, 5, 1)
```
Comment (mathstuf) [inline src/commands.c]: Might want to look at bumping our minimum version up to something like 2.6 or so (in Debian stable; unstable has 2.10).

File: src/commands.c [hunk 3]
```diff
@@ -598,19 +585,10 @@ parse_command_from_file (const char *cmd)
 
 /* ========================= COMMAND TABLE ========================== */
 
-#ifdef USE_WEBKIT2
 #if WEBKIT_CHECK_VERSION (1, 11, 4)
 #define HAVE_PLUGIN_API
 #endif
 #define HAVE_SECURITY
```
Comment (mathstuf) [inline src/commands.c]: This and the plugin API seem to always be true, so simplifying based on that is possible.

File: src/commands.c (hunk 2 above)
Comment (mathstuf) [inline src/commands.c]: And we definitely need 2.4 (for [this function](http://webkitgtk.org/reference/webkit2gtk/stable/WebKitWebContext.html#webkit-web-context-set-web-extensions-initialization-user-data)).

File: src/commands.c [hunk 4]
```diff
@@ -1134,97 +976,20 @@ IMPLEMENT_COMMAND (cookie)
...
-        soup_cookie_jar_add_cookie (SOUP_COOKIE_JAR (uzbl.net.soup_cookie_jar), cookie);
-        uzbl.net.soup_cookie_jar->in_manual_add = 0;
-
-        if (expires) {
-            soup_date_free (expires);
-        }
-#endif
     } else if (!g_strcmp0 (command, "delete")) {
-#ifdef USE_WEBKIT2
         uzbl_debug ("Manual cookie deletions are unsupported in WebKit2.\n");
```
Comment (mathstuf) [inline src/commands.c]: Ugh. We'll have to look into how to do this; uzbl won't be nearly as safe with its cookies in WebKit2 as-is. Probably a new issue.

File: src/commands.c [hunk 5]
```diff
@@ -1072,57 +964,7 @@ IMPLEMENT_COMMAND (save)
     }
 }
 #endif
-#endif
-
-#ifndef USE_WEBKIT2
-IMPLEMENT_COMMAND (frame)
```
Comment (mathstuf) [inline src/commands.c]: This one may be implementable using some extension code, but not sure. Should be a new issue.

File: src/gui.c [hunk 6]
```diff
@@ -371,22 +316,13 @@ web_view_init ()
...
         "signal::button-release-event",                 G_CALLBACK (button_release_cb),        NULL,
-#ifdef USE_WEBKIT2
         "signal::mouse-target-changed",                 G_CALLBACK (mouse_target_cb),          NULL,
-#else
-        "signal::hovering-over-link",                   G_CALLBACK (link_hover_cb),            NULL,
-#endif
         /* Page metadata events */
         "signal::notify::title",                        G_CALLBACK (title_change_cb),          NULL,
-#ifdef USE_WEBKIT2
         "signal::notify::estimated-load-progress",
```
Comment (mathstuf) [inline src/gui.c]: The next line can be collapsed into this one.

File: src/gui.c [hunk 7]
```diff
@@ -1983,13 +1651,6 @@ get_click_context ()
         return NO_CLICK_CONTEXT;
     }
 
-#ifndef USE_WEBKIT2 /* TODO: No API for this? :( */
```
Comment (mathstuf) [inline src/gui.c]: It looks like the extension has access to this. New issue.

File: src/inspector.c [hunk 8]
```diff
@@ -46,33 +37,13 @@ uzbl_inspector_init ()
...
 
     g_object_connect (G_OBJECT (uzbl.gui.inspector),
-#ifdef USE_WEBKIT2
         "signal::bring-to-front",        G_CALLBACK (inspector_create_cb),              NULL,
-#else
-        "signal::inspect-web-view",      G_CALLBACK (inspector_create_cb),              NULL,
-        "signal::show-window",           G_CALLBACK (inspector_show_window_cb),         NULL,
-#endif
-#ifdef USE_WEBKIT2
         "signal::closed",
```
Comment (mathstuf) [inline src/inspector.c]: This and the next few signals can be un-split.

**Comment by mathstuf** [comment]
Some whitespace errors and stray comments are left from the line removals. Can't comment on `variables.c` here due to it being "too big". The config file should be cleaned up to not use variables which no longer exist (`proxy_url`, `http_debug`, `ssl_ca_file`, etc.). The EM might need some changes to remove references to those variables. Some other variables are less useful now (`WEBKIT_UA_MAJOR`, `HAS_WEBKIT2`, etc.). The now-pointless typedefs in `webkit.h` can also be factored over the rest of the codebase as well.

If you want me to tackle this, I can try it out tonight.

**Comment by keis** [comment]
Due the size of this pull request I was thinking it would be better to have GTK3 and the version bump as separate PRs to make them easier to review. I'll make sure to clean up the noise from this one though with whitespace and documentation.

If you want to rip out gtk2 or checks for old version that would be cool :)

**Comment by keis** [comment]
This is the list of removed variables

enable_builtin_auth
auto_resize_window
proxy_url
max_conns
max_conns_host
http_debug
ssl_ca_file
enable_universal_file_access
enable_cross_file_access
display_insecure_content
maintain_history
profile_js
profile_timeline
transparent
window_view_mode
javascript_dom_paste
autoshrink_images
use_image_orientation
editing_behavior
enforce_95_dpi
minimum_font_size
minimum_logical_font_size
enable_3d_acceleration
enable_css_shaders
app_cache_size
web_database_directory
current-encoding

**Comment by keis** [comment]
The network related of those would be nice to have again (proxy_url, max_conns, ssl_ca_file)

**Comment by keis** [comment]
Closing in favor of #234

**Pull request closed without merging**